framework/include/constraints/ElemElemConstraint.h

```diff
@@ -59,10 +59,17 @@ class ElemElemConstraint : public Constraint,
    */
   virtual void computeJacobian();
 
+  /**
+   * Get the interface ID
+   */
+  virtual const unsigned int & getInterfaceID() const { return _interface_id; };
```

Comment:
Minor: const ref to an unsigned int is overkill. You might as well just return the unsigned int by value. Also, do you really want this to be virtual?

Reply:
I need to override this function for XFEM cases where the interface id will be obtained directly from the mesh cutters.

Reply:
I just realized that it could be  not virtual :-)